controlplane/kubeadm/controllers/controller_test.go

```diff
@@ -1222,12 +1209,17 @@ func (c *fakeClient) Create(ctx context.Context, obj runtime.Object, opts ...cli
 }
 
 func createClusterWithControlPlane() (*clusterv1.Cluster, *controlplanev1.KubeadmControlPlane, *unstructured.Unstructured) {
-	cluster := newCluster(&types.NamespacedName{Name: "foo", Namespace: "test"})
+	kcpName := fmt.Sprintf("kcp-foo-%s", util.RandomString(6))
+
+	namespace := "test"
+	cluster := newCluster(&types.NamespacedName{Name: kcpName, Namespace: namespace})
+	cluster.UID = "1"
 	cluster.Spec = clusterv1.ClusterSpec{
 		ControlPlaneRef: &corev1.ObjectReference{
 			Kind:       "KubeadmControlPlane",
-			Namespace:  "test",
-			Name:       "kcp-foo",
+			Namespace:  namespace,
+			Name:       kcpName,
+			UID:        "1",
```

Comment:
As above, these UIDs won't be correct. I'm wondering if we need them to be correct for these tests, or we just need them to be some value?

Reply:
We only need it for kcp owner reference otherwise getting this error: metadata.ownerReferences.uid: Invalid value: \"\": uid must not be empty"

The rest was unneccessary.